controllers/datadogagent/override/global.go

```diff
@@ -6,34 +6,220 @@
+	manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+		Name:  apicommon.DDLogLevel,
+		Value: *config.LogLevel,
+	})
+
+	// NetworkPolicy contains the network configuration.
+	if config.NetworkPolicy != nil {
+		if apiutils.BoolValue(config.NetworkPolicy.Create) {
+			switch config.NetworkPolicy.Flavor {
+			case v2alpha1.NetworkPolicyFlavorKubernetes:
+				err = resourcesManager.NetworkPolicyManager().AddKubernetesNetworkPolicy(component.BuildKubernetesNetworkPolicy(dda, componentName))
+			case v2alpha1.NetworkPolicyFlavorCilium:
+				// TODO
+			}
+			if err != nil {
+				logger.Info("Error adding Network Policy to the store", "error", err)
+			}
 		}
-		manager.EnvVar().AddEnvVarToContainer(apicommonv1.CoreAgentContainerName, kubeletEnvVar)
-		manager.EnvVar().AddEnvVarToContainer(apicommonv1.ProcessAgentContainerName, kubeletEnvVar)
 	}
 
-	// set image registry
+	if componentName == v2alpha1.NodeAgentComponentName {
+		// Tags contains a list of tags to attach to every metric, event and service check collected.
+		if config.Tags != nil {
+			tags, marshalErr := json.Marshal(config.Tags)
+			if marshalErr != nil {
+				logger.Info("Failed to unmarshal json input", "error", err)
+			} else {
+				manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+					Name:  apicommon.DDTags,
+					Value: string(tags),
+				})
+			}
+		}
+
+		// Provide a mapping of Kubernetes Labels to Datadog Tags.
+		if config.PodLabelsAsTags != nil {
+			podLabelsAsTags, marshalErr := json.Marshal(config.PodLabelsAsTags)
+			if marshalErr != nil {
+				logger.Info("Failed to unmarshal json input", "error", err)
+			} else {
+				manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+					Name:  apicommon.DDPodLabelsAsTags,
+					Value: string(podLabelsAsTags),
+				})
+			}
+		}
+
+		// Provide a mapping of Kubernetes Annotations to Datadog Tags.
+		if config.PodAnnotationsAsTags != nil {
+			podAnnotationsAsTags, marshalErr := json.Marshal(config.PodAnnotationsAsTags)
+			if marshalErr != nil {
+				logger.Info("Failed to unmarshal json input", "error", err)
+			} else {
+				manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+					Name:  apicommon.DDPodAnnotationsAsTags,
+					Value: string(podAnnotationsAsTags),
+				})
+			}
+		}
+
+		// LocalService contains configuration to customize the internal traffic policy service.
+		gitVersion := resourcesManager.Store().GetVersionInfo()
+		if utils.IsAboveMinVersion(gitVersion, minLocalServiceVersion) {
+			if utils.IsAboveMinVersion(gitVersion, minDefaultLocalServiceVersion) || (config.LocalService != nil && apiutils.BoolValue(config.LocalService.ForceEnableLocalService)) {
+				if config.LocalService != nil && config.LocalService.NameOverride != nil {
+					err = resourcesManager.ServiceManager().AddService(component.BuildAgentLocalService(dda, *config.LocalService.NameOverride))
+				} else {
+					err = resourcesManager.ServiceManager().AddService(component.BuildAgentLocalService(dda, ""))
+				}
+				if err != nil {
```

Comment:
This could evaluate to true if we had an error in a previous block of code.
Might be worth to define a new `errAddService` or similar to avoid confusion.